Implement applyMiddlewares

mkaulig opened the pull request:
Implement a method applyMiddlewares inspired by the redux function with
the same name. This method returns an enhancer, which can enhance store
creators to execute the applied middlewares before normal dispatch
calls.

mkaulig committed:
Implement applyMiddlewares

Implement a method applyMiddlewares inspired by the redux function with
the same name. This method returns an enhancer, which can enhance store
creators to execute the applied middlewares before normal dispatch
calls.

glung commented on lib/src/main/java/com/glung/redux/Middlewares.java:
```diff
@@ -0,0 +1,75 @@
+
+        @Override
+        public Object dispatch(Object action) {
+            return createMiddlewareDispatcher(middlewares).dispatch(action);
```
Do you think we could create the Dispatcher it only once by calling `createMiddlewareDispatcher` in `applyMiddlewares` ?

glung reviewed (changes requested):
Sorry totally missed your PR ! Thx for contributing 

Could you add tests ?

AngusMorton commented on lib/src/main/java/com/glung/redux/Middlewares.java:
```diff
@@ -0,0 +1,75 @@
+            return createMiddlewareDispatcher(middlewares).dispatch(action);
+        }
+
+        private Dispatcher createMiddlewareDispatcher(final List<redux.api.enhancer.Middleware<S>> middlewares) {
```
This method should be static

mkaulig committed:
Create Middleware Dispatchers On Store Creation

Create middleware dispatchers in the middleware enhancer only once
during store creation

mkaulig commented:
Thanks for the feedback! I added some Tests and changed the implementation to create the ```Dispatcher``` only once in ```applyMiddlewares```

glung commented:
Hey @mkaulig just a message to say I am not ignoring this PR, it will look into it asap

glung reviewed (approved):
Thx! I am just going to apply some minor style related changes on master after landing it
